compiler/src/dotty/tools/dotc/typer/Checking.scala

```diff
@@ -886,11 +886,11 @@ trait Checking {
    */
   def checkNoModuleClash(sym: Symbol)(using Context): Unit =
     if sym.effectiveOwner.is(Package)
-       && sym.owner.info.member(sym.name.moduleClassName).symbol.isAbsent()
+       && sym.effectiveOwner.info.member(sym.name.moduleClassName).symbol.isAbsent()
```

Comment:
While typing `val O1: O1 = new O1()`, if we use owner then we run back into the enclosing `aa1` val def which is completing.  Somehow that cyclic error leads to a type leak and thus a crash, but this is the root cause.  WDYT?

Reply:
That change looks reasonable to me, I would assign `sym.effectiveOwner` to a val instead of calling it repeatedly.